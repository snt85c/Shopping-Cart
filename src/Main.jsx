import { Routes, Route, HashRouter } from "react-router-dom";
import { useEffect, useState } from "react";
import "./index.css";
import Navbar from "./Navbar";
import Cart from "./Cart";
import SuggestionPage from "./SuggestionPage";
import AttractionPage from "./AttractionPage";
import EventPage from "./EventPage";
import NoPath from "./NoPath";
import Alert from "./AlertComponents/Alert";
import { AlertContextProvider } from "./AlertComponents/AlertContextProvider";

export default function Main() {
  let [onAttractionScreen, setOnAttractionScreen] = useState([]);
  let [onEventScreen, setOnEventScreen] = useState([]);
  let [cart, setCart] = useState({
    display: "none",
    items: [],
  });

  useEffect(() => {
    if (localStorage.getItem("cart")) {
      setCart(JSON.parse(localStorage.getItem("cart")));
    }
  }, []);

  useEffect(() => {
    if (localStorage.getItem("onAttractionScreen")) {
      setOnAttractionScreen(
        JSON.parse(localStorage.getItem("onAttractionScreen"))
      );
    }
  }, []);

  useEffect(() => {
    if (localStorage.getItem("onEventScreen")) {
      setOnEventScreen(JSON.parse(localStorage.getItem("onEventScreen")));
    }
  }, []);

  useEffect(() => {
    localStorage.setItem("cart", JSON.stringify(cart));
  }, [cart]);

  useEffect(() => {
    localStorage.setItem(
      "onAttractionScreen",
      JSON.stringify(onAttractionScreen)
    );
  }, [onAttractionScreen]);

  useEffect(() => {
    localStorage.setItem("onEventScreen", JSON.stringify(onEventScreen));
  }, [onEventScreen]);

  return (
    <>
      <AlertContextProvider>
        <HashRouter>
          <Alert />
          <Navbar
            cart={cart}
            setCart={setCart}
            setOnScreen={setOnAttractionScreen}
          />
          <Cart cart={cart} setCart={setCart} />
          <Routes>
            <Route
              path="/"
              element={<SuggestionPage setOnScreen={setOnAttractionScreen} />}
            />
            <Route
              path=":second"
              element={
                <AttractionPage
                  onScreen={onAttractionScreen}
                  setOnScreen={setOnEventScreen}
                />
              }
            />
            <Route
              path=":second/:third"
              element={
                <EventPage
                  onScreen={onEventScreen}
                  cart={cart}
                  setCart={setCart}
                />
              }
            />

            <Route path="*" element={<NoPath />} />
          </Routes>
        </HashRouter>
      </AlertContextProvider>
    </>
  );
}
