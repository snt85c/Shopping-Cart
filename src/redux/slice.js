import { createSlice } from "@reduxjs/toolkit";

export const shopSlice = createSlice({
  name: "counter",
  initialState: {
    isDarkMode: false,
    suggestions: [],
    artist: {},
    events: [],
    venue: {},
    artistTopTracks: {},
    artistMetadata: {},
    cart: {
      items: [],
      isCheckoutClicked: false,
    },
  },
  reducers: {
    setIsDarkMode: (state, actions) => {
      state.isDarkMode = actions.payload;
    },
    setSuggestions: (state, actions) => {
      state.suggestions = actions.payload;
    },
    setArtist: (state, actions) => {
      state.artist = actions.payload;
    },
    setEvents: (state, actions) => {
      state.events = actions.payload;
    },
    setVenue: (state, actions) => {
      state.venue = actions.payload;
    },
    setArtistTopTracks: (state, actions) => {
      state.artistTopTracks = actions.payload;
    },
    setArtistMetadata: (state, actions) => {
      state.artistMetadata = actions.payload;
    },
    setCart: (state, actions) => {
      switch (actions.payload.type) {
        case "ADD_TO_CART":
          state.cart.items.push(actions.payload.payload);
          break;
        case "INCREASE_ITEM":
          const objectsIncr = [...state.cart.items];
          objectsIncr[actions.payload.index].ticketInCart++;
          state.cart.items = objectsIncr;
          break;
        case "DECREASE_ITEM":
          const objectsDecr = [...state.cart.items];
          objectsDecr[actions.payload.index].ticketInCart--;
          state.cart.items = objectsDecr;
          break;
        case "REMOVE_ITEM":
          state.cart.items = state.cart.items.filter(
            (item) => item.id !== actions.payload.payload.id
          );
          break;
        case "CHECKOUT_CART":
          state.cart.items = [];
          state.cart.isCheckoutClicked = true;
          break;
      }
    },
  },
});

export const {
  setSuggestions,
  setArtist,
  setIsDarkMode,
  setCart,
  setEvents,
  setVenue,
  setArtistMetadata,
  setArtistTopTracks,
} = shopSlice.actions;
