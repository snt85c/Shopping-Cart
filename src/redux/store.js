import { configureStore } from "@reduxjs/toolkit";
import { shopSlice} from "./slice";

export default configureStore({
  reducer: shopSlice,
});

