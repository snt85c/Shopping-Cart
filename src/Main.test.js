import { render, cleanup, screen } from "@testing-library/react";
import { store } from "./redux/store";
import { Provider } from "react-redux";
import "matchmedia-polyfill";

import Main from "./Main";



describe("main test", () => {
  it("renders Apps", () => {
    const { container } = render(
      <Provider store={store}>
        <Main />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("render ticketmaster ", () => {
    render(
      <Provider store={store}>
        <Main />
      </Provider>
    );
    const ticketmasterElement = screen.getAllByText(/Ticketmaster/i);
    expect(ticketmasterElement).toHaveLength(2);
  });
});
