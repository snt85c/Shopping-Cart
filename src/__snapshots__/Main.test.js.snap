// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main test renders Apps 1`] = `
<div
  class=" w-full relative top-0 flex  flex-col md:flex-row justify-between items-center max-w-full border-b-amber-500 dark:bg-gray-900 bg-white border-b-2 font-extrabold text-4xl  "
>
  <div
    class="flex w-full md:justify-start pt-2 md:pt-0 md:pl-2 font-newake subpixel-antialiased tracking-wider justify-center cursor-pointer dark:text-white duration-1000 text-black "
  >
    Ticketmaster
  </div>
  <div
    class="flex w-full md:justify-end md:mr-10 justify-center items-center"
  >
    <div
      class="form-control"
    >
      <svg
        class="m-1"
        color="black"
        fill="none"
        height="25"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="cursor: pointer; transform: rotate(90deg);"
        viewBox="0 0 24 24"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <mask
          id="circle-mask-1"
        >
          <rect
            fill="white"
            height="100%"
            width="100%"
            x="0"
            y="0"
          />
          <circle
            fill="black"
            r="9"
          />
        </mask>
        <circle
          cx="12"
          cy="12"
          fill="black"
          mask="url(#circle-mask-1)"
        />
        <g
          stroke="currentColor"
          style="opacity: 1;"
        >
          <line
            x1="12"
            x2="12"
            y1="1"
            y2="3"
          />
          <line
            x1="12"
            x2="12"
            y1="21"
            y2="23"
          />
          <line
            x1="4.22"
            x2="5.64"
            y1="4.22"
            y2="5.64"
          />
          <line
            x1="18.36"
            x2="19.78"
            y1="18.36"
            y2="19.78"
          />
          <line
            x1="1"
            x2="3"
            y1="12"
            y2="12"
          />
          <line
            x1="21"
            x2="23"
            y1="12"
            y2="12"
          />
          <line
            x1="4.22"
            x2="5.64"
            y1="19.78"
            y2="18.36"
          />
          <line
            x1="18.36"
            x2="19.78"
            y1="5.64"
            y2="4.22"
          />
        </g>
      </svg>
    </div>
    <div
      class="absolute top-2 right-3 cursor-pointer text-amber-500"
    >
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
    <input
      class="flex m-2 top-36 dark:bg-black bg-white dark:text-white duration-[1000ms] text-black text-sm p-2 mx-2 w-2-3 rounded-md  border-amber-500 border-2"
      placeholder="Search by Artist"
    />
    <div
      class=" absolute flex flex-col z-20  right-auto top-24  md:right-12 md:top-14"
      style="display: none;"
    />
    <div
      class="absolute top-1 md:top-2 right-3 md:rigth-2 w-5 h-5 text-center  text-xs font-bold z-10 rounded-full border-2 border-amber-500 dark:bg-gray-900  bg-white text-red-500"
    >
      0
    </div>
  </div>
  <div
    class="cartOverlay absolute shadow-lg z-20 min-h-[90vh] top-[104px] md:top-[56px] right-0  dark:bg-gradient-to-t from-amber-500 to-gray-900 duration-1000 bg-white text-black dark:text-white w-2/4 md:w-1/3  text-center text-sm border-amber-500 border-l-2 border-b-2 pt-2  cartAnimation"
    style="display: none;"
  >
    <div
      class="flex flex-col justify-center items-center fadeInAnimation"
    >
       
      <div
        class="my-6 font-extrabold"
      >
        Your cart is empty at the moment
      </div>
      <img
        alt="empty_cart"
        class="w-1/2 h-1/2 dark:invert opacity-50 duration-1000"
        src="empty_cart.webp"
      />
    </div>
  </div>
</div>
`;
